main.go

```diff
@@ -134,17 +139,21 @@ func androidTestVariantPairs(module string, variantsMap gradle.Variants) (gradle
 }
 
 func mainE(config Configs) error {
-	gradleProject, err := gradle.NewProject(config.ProjectLocation)
+	gradleProject, err := gradle.NewProject(config.ProjectLocation, cmdFactory)
 	if err != nil {
 		return fmt.Errorf("Failed to open project, error: %s", err)
 	}
 
-	buildTask := gradleProject.
-		GetTask("assemble")
+	buildTask := gradleProject.GetTask("assemble")
+
+	args, err := shellquote.Split(config.Arguments)
```

Comment:
Moved this section up a bit to be able to use the parsed args in the `GetVariants()` call